lib/github_service/commands/cross_repo_test.rb

```diff
@@ -187,18 +187,31 @@ def dedup_repo_list(repo_list)
           .sort # Unadorned repo name will always sort before adorned repo names
           .uniq
           .slice_when { |a, b| bare_repo_name(a) != bare_repo_name(b) }
-          .map(&:last)
+          .map do |repos|
+            repos.shift if repos.size > 1 && bare_repo_name?(repos.first)
+            repos.size == 1 ? repos.first : repos
+          end
       end
 
       def bare_repo_name(repo)
         repo.split(/[@#]/, 2).first
       end
 
+      def bare_repo_name?(repo)
```

Comment:
I wonder if `bare_repo?` would be a better name since this is the inverse of `branch_or_pr?` ?  It looks confusing that there is a `bare_repo_name` and a `bare_repo_name?` that are unrelated.